### PR TITLE
Rollout even if workload revision is not changed

### PR DESCRIPTION
**Pull request by veophi**
Signed-off-by: mingzhou.swx <mingzhou.swx@alibaba-inc.com>

### Ⅰ. Describe what this PR does
You can trigger rollout  manually via adding an annotation `rollouts.kruise.io/in-progressing: "true"` to Workload, even if workload revision is not changed.

### CODE REVIEW

File: test/e2e/rollout_test.go [hunk 1]
```diff
@@ -3663,6 +3663,96 @@ var _ = SIGDescribe("Rollout", func() {
 			CheckPodBatchLabel(workload.Namespace, workload.Spec.Selector, "2", "4", 1)
 			CheckPodBatchLabel(workload.Namespace, workload.Spec.Selector, "2", "5", 1)
 		})
+
+		It("patch batch id to pods: only rollout-id changes", func() {
```
Comment (furykerry) [inline test/e2e/rollout_test.go:3667]: add a case for rolling to 20%, and rollout change rollout id

File: pkg/controller/batchrelease/workloads/deployment_canary_control_plane.go [hunk 2]
```diff
@@ -62,46 +62,10 @@ func NewDeploymentRolloutController(cli client.Client, recorder record.EventReco
 
 // VerifyWorkload verifies that the workload is ready to execute release plan
 func (c *DeploymentsRolloutController) VerifyWorkload() (bool, error) {
```
Comment (furykerry) [inline pkg/controller/batchrelease/workloads/deployment_canary_control_plane.go:64]: is the VerifyWorkload needed after the changes ? If yes, the name "VerifyWorkload" seems improper anymore

Comment (furykerry) [inline pkg/controller/batchrelease/workloads/deployment_canary_control_plane.go:64]: is the VerifyWorkload needed after the changes ? If yes, the name "VerifyWorkload" seems improper now.

File: test/e2e/rollout_test.go (hunk 1 above)
Comment (veophi) [reply test/e2e/rollout_test.go:3667]: I rolling 60%, and change rollout-id in `Line 3731`.

Comment (veophi) [reply test/e2e/rollout_test.go:3667]: done

File: pkg/controller/batchrelease/workloads/deployment_canary_control_plane.go (hunk 2 above)
Comment (veophi) [reply pkg/controller/batchrelease/workloads/deployment_canary_control_plane.go:64]: we will remove it in near future.

**Review by furykerry — COMMENTED** [review]
/lgtm

File: pkg/controller/rollout/canary.go [hunk 3]
```diff
@@ -45,13 +45,21 @@ func (r *rolloutContext) runCanary() error {
...
 		canaryStatus.CanaryReadyReplicas = r.workload.CanaryReadyReplicas
 	} else {
+		canaryStatus.Message = fmt.Sprintf("BatchRelease at state %s, id %s, step %d",
+			batch.Status.CanaryStatus.CurrentBatchState, batch.Status.ObservedRolloutID, batch.Status.CanaryStatus.CurrentBatch+1)
 		canaryStatus.CanaryReplicas = batch.Status.CanaryStatus.UpdatedReplicas
 		canaryStatus.CanaryReadyReplicas = batch.Status.CanaryStatus.UpdatedReadyReplicas
 	}
 
+	// sync rollout-id to batchRelease if we need
+	if err := r.batchControl.SyncRolloutID(getRolloutID(r.workload, r.rollout)); err != nil {
```
Comment (zmberg) [inline pkg/controller/rollout/canary.go:59]: What is the canaryStatus.RolloutId not sync? I think it should sync from canaryStatus.RolloutId to br.

Comment (veophi) [reply pkg/controller/rollout/canary.go]: done

**Pull request merged**